plugins/module_utils/network/sonic/config/stp/stp.py

```diff
@@ -220,16 +219,25 @@ def _state_replaced(self, want, have, diff):
         """
         commands = []
         mod_commands = []
+        tmp_have = deepcopy(have)
         replaced_config, requests = self.get_replaced_config(want, have)
 
         if replaced_config:
             commands.extend(update_states(replaced_config, "deleted"))
             mod_commands = want
+
+            for command in commands:
+                self.transform_config_for_diff_check(command)
+            self.transform_config_for_diff_check(tmp_have)
+            new_config = get_new_config(commands, tmp_have, TEST_KEYS_generate_config)
+            new_config = self.post_process_generated_config(new_config)
+            tmp_have = new_config
```

Comment:
Instead of using all of the commands in "want" for mod_commands for this case, it would be more efficient to set "mod_commands = get_diff(want, tmp_have, TEST_KEYS)"

Reply:
Thanks for adding this additional enhancement.

Resolved.